### PR TITLE
Added documentation for plotting

### PR DESCRIPTION
**Pull request by josephineyates**
Updated the documentation so it fit the new plotting utilities + updated the description of the output folder, which was outdated

### CODE REVIEW

**Comment by josephineyates** [comment]
@pawel-czyz I am getting this error for the pytype check that I do not understand: 

```
FAILED: /home/runner/work/CanSig/CanSig/.pytype/pyi/cansig/run/postprocessing.pyi 
/opt/hostedtoolcache/Python/3.8.12/x64/bin/python -m pytype.single --imports_info /home/runner/work/CanSig/CanSig/.pytype/imports/cansig.run.postprocessing.imports --module-name cansig.run.postprocessing --platform linux -V 3.8 -o /home/runner/work/CanSig/CanSig/.pytype/pyi/cansig/run/postprocessing.pyi --analyze-annotated --enable-nested-classes --nofail --quick /home/runner/work/CanSig/CanSig/cansig/run/postprocessing.py
File "/home/runner/work/CanSig/CanSig/cansig/run/postprocessing.py", line 10, in <module>: Can't find module 'cansig.plotting'. [import-error]
```

This is surprising because I have only changed the documentation in this PR and the main branch passes the build (from which I spawned this one). Also, cansig.plotting exists... Do you have any clue what I can do?

**Comment by pawel-czyz** [comment]
> @pawel-czyz I am getting this error for the pytype check that I do not understand:
> 
> ```
> FAILED: /home/runner/work/CanSig/CanSig/.pytype/pyi/cansig/run/postprocessing.pyi 
> /opt/hostedtoolcache/Python/3.8.12/x64/bin/python -m pytype.single --imports_info /home/runner/work/CanSig/CanSig/.pytype/imports/cansig.run.postprocessing.imports --module-name cansig.run.postprocessing --platform linux -V 3.8 -o /home/runner/work/CanSig/CanSig/.pytype/pyi/cansig/run/postprocessing.pyi --analyze-annotated --enable-nested-classes --nofail --quick /home/runner/work/CanSig/CanSig/cansig/run/postprocessing.py
> File "/home/runner/work/CanSig/CanSig/cansig/run/postprocessing.py", line 10, in <module>: Can't find module 'cansig.plotting'. [import-error]
> ```
> 
> This is surprising because I have only changed the documentation in this PR and the main branch passes the build (from which I spawned this one). Also, cansig.plotting exists... Do you have any clue what I can do?

Currently the structure looks like `cansig/plotting/plotting.py` and `cansig/plotting` is _not_ a subpackage (doesn't have `__init__.py`), so technically you can't import stuff from it. ((a) A "smoke" unit test would perhaps find that, (b) There are Python implicit module conventions, so it may work even without `__init__.py`, but I had very mixed experiences with them).

File: docs/index.rst [hunk 1]
```diff
@@ -86,22 +86,21 @@ This is one of many databases available online. Alternatively, one can provide a
...
-    * ``training_status.json``: summary whether all the dimension reduction/batch correction methods have converged
-* ``runs/``: directory with all generated data, from single runs
-    * ``latent/``: each model has a separate directory representing the latent representations inferred by the batch correction/dimension reduction method
-    * ``cluster/``: cluster assignments (each of the latent representations file can clustered using different algorithms)
-    * ``gsea/``: Gene Set Enrichment Analysis scores for each of the clustering
-
-.. todo::
-   This is a sketch. We need to adjust it accordingly, when the implementation is ready.
+* ``integration/``: each model has a separate directory representing the latent representations inferred by the batch correction/dimension reduction method
+   * ``{rundir}/integration-settings.json``: summary of the parameters used for the integration method
```
Comment (pawel-czyz) [inline docs/index.rst:94]: In our case `{rundir}` is `tutorial-output`.

File: docs/index.rst [hunk 2]
```diff
@@ -86,22 +86,21 @@ This is one of many databases available online. Alternatively, one can provide a
 .. todo::
    Fill in the TODO items above.
 
-The results will be available in the generated directory ``tutorial-output/`` in a few minutes.
+The results will be available in the generated directory ``tutorial-output/`` in a few minutes to hours depending on the size of the dataset.
```
Comment (pawel-czyz) [inline docs/index.rst:89]: In our case the dataset is small to make sure it takes a few minutes.

File: docs/index.rst (hunk 1 above)
Comment (josephineyates) [reply docs/index.rst:94]: Are you sure? Here the rundir would correspond to something like `20220524-114857-huge-gopher`, wouldn't that also be the case for the tutorial? tutorial_output would be the full directory with the results, wouldn't it?

File: docs/index.rst (hunk 2 above)
Comment (pawel-czyz) [reply docs/index.rst:89]: I really think that we should give an example which can be analysed in 10 minutes on a CPU – I don't think I'd have enough patience to wait ~3 hours to see whether the tool is useful or not.

File: docs/index.rst (hunk 1 above)
Comment (pawel-czyz) [reply docs/index.rst:94]: Hmmm, good point – we have a hierarchical structure. I.e., we have integration runs and postprocessing runs and inside them we have different `{rundir}`s.

File: docs/pipeline-advanced.rst [hunk 3]
```diff
@@ -39,3 +39,43 @@ When the directory with different latent codes is ready, run:
...
+* ``--sigcols``: optionally, the name of the signatures to plot in addition to batch and cluster assignment. 
+* ``--disable-plots``: if you do not want plots to be generated, use this flag. This will skip plotting altogether.
+.. note::
+   The signature column names inputted with ``--sigcols`` must be in the adata.obs.columns.
+
+Example usage to plot UMAP with additional scored signatures named ``signature1`` and ``signature2``.
+
+.. code-block:: bash
+
+   $ python -m cansig.run.pipeline data/pipeline-tutorial/data.hdf5
```
Comment (pawel-czyz) [inline docs/pipeline-advanced.rst:62]: @FlorianBarkmann @josephineyates Do we already have a dataset where these signatures are present? If not, it'd be good to open an issue, so we don't forget.

File: docs/index.rst (hunk 2 above)
Comment (josephineyates) [reply docs/index.rst:89]: Yes, I resolved because I removed the modif ;)

File: docs/pipeline-advanced.rst (hunk 3 above)
Comment (josephineyates) [reply docs/pipeline-advanced.rst:62]: I asked @FlorianBarkmann to create one, he should be on it, but you can always add an issue :)

**Pull request merged**